modules/broker-auth/src/main/java/org/wso2/broker/auth/AuthManager.java

```diff
@@ -0,0 +1,164 @@
+     */
+    private void registerSaslServers() {
+        // create PLAIN SaslServer builder
+        PlainSaslServerBuilder plainSaslServerBuilder = new PlainSaslServerBuilder();
+        saslMechanisms.put(plainSaslServerBuilder.getMechanismName(), plainSaslServerBuilder);
+        // Register given Sasl Server factories
+        if (Security
+                .insertProviderAt(new BrokerSecurityProvider(BrokerAuthConstants.PROVIDER_NAME, saslMechanisms), 1)
+                == -1) {
+            LOGGER.info("AMQ security authentication providers are already installed.");
+        } else {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("AMQ security authentication mechanisms providers are successfully registered.");
+            }
+        }
+    }
+
+    /**
+     * Creates Jaas config
+     *
+     * @param loginModuleClassName Jaas login module class name
+     * @return Configuration
+     */
+    private static Configuration createJaasConfig(String loginModuleClassName, UserStoreManager userStoreManager,
+                                                  Map<String, Object> options
+                                                 ) {
+        options.put(BrokerAuthConstants.USER_STORE_MANAGER_PROPERTY, userStoreManager);
+        AppConfigurationEntry[] entries = {
+                new AppConfigurationEntry(loginModuleClassName, AppConfigurationEntry.LoginModuleControlFlag.REQUIRED,
+                                          options)
+        };
+        return new Configuration() {
+            @Override
+            public AppConfigurationEntry[] getAppConfigurationEntry(String name) {
+                return entries;
+            }
+        };
+    }
+
+    /**
+     * Create sasl server for given mechanism
+     *
+     * @param hostName  Hostname of the server
+     * @param mechanism Sasl mechanism
+     * @return Sasl server created for mechanism
+     * @throws SaslException Throws if server does not support for given mechanism
+     */
+    public SaslServer createSaslServer(String hostName, String mechanism) throws SaslException {
+        SaslServerBuilder saslServerBuilder = saslMechanisms.get(mechanism);
+        if (saslServerBuilder != null) {
+            return Sasl.createSaslServer(mechanism, BrokerAuthConstants.AMQP_PROTOCOL_IDENTIFIER, hostName,
+                                         saslServerBuilder.getProperties(),
+                                         saslServerBuilder.getCallbackHandler());
+        } else {
+            throw new SaslException("Server does not support for mechanism: " + mechanism);
+        }
+    }
+
+    /**
+     * Authenticate response based on given sasl server
+     *
+     * @param saslServer Sasl server
+     * @param response   Client response
+     * @return challenge
+     * @throws SaslException Throws if error occurs while evaluating the response
+     */
+    public byte[] authenticate(SaslServer saslServer, byte[] response) throws SaslException {
```

Comment:
Do we need this method in auth manager? Can we do saslServer.evaluateResponse(response) in respective class?